prover/src/lib.rs

```diff
@@ -67,10 +67,11 @@ impl BlockProverInput {
             let block_number = block_number
                 .to_u64()
                 .context("block number overflows u64")?;
-            let prev = previous.map(|p| GeneratedBlockProof {
-                b_height: block_number - 1,
-                intern: p,
-            });
+            let prev = if let Some(prev) = previous {
+                Some(prev.await?)
+            } else {
+                None
+            };
```

Comment:
```suggestion
            let prev = match prev {
                  Some(it) => Some(it.await?),
                  None => None
            };
```
nit: why say lot word when few word do trick